source/codegen/metadata_validation.py

```diff
@@ -228,7 +231,11 @@ def validate_function(function_name: str, metadata: dict):
                     ivi_dance_with_a_twist_params.append(parameter['name'])
                     ivi_dance_with_a_twist_sizes.add(size['value'])
                     ivi_dance_with_a_twist_twists.add(size['value_twist'])
-
+                if disallow_session_reverse_lookup:
+                    if not is_init_method:
+                        if parameter['direction'] == 'out' and parameter['type'] == service_helpers.get_resource_handle_type(metadata['config']):
+                            raise  Exception(
+                               f"Since reverse session lookup is disallowed for this driver, {function_name} is not valid since it's not marked as an init_method but has a session output!")
```

Comment:
I don't see anything in the code about "why" reverse lookup would be disallowed. Either from the perspective of someone adding a new driver (that might want to toggle that flag) or someone trying to understand a driver that disallows it (not sure why they would care, but could happen).

Some ideas:
* Comment in metadata.
* Update this message saying why reverse lookup is disallowed.
* Update the name of the flag to be something like "duplicate_resource_handles_allowed" which implies reverse lookup is not allowed.
* Comments in tests.

I think putting stuff in this file is probably the most useful. Even though the schema is not exactly purely declarative, it does specifically exist to explain what the metadata means. So it's a good place for more info.

Reply:
Good point. I went with your second and third idea. Change the flag to 'duplicate_resource_handles_allowed' and update the Exception message in validation to explain why  it failed.

For rfsg, the new message would look like
```
Exception: nirfsg allows duplicate resource handles in the session repository. Therefore, the handle we'd get for "value" can't be mapped back to a Session to provide to the client!
...
Exception: Failed to validate function GetAttributeViSession
```
which I think gives a good idea of what's going on.